lib/linguist/heuristics.rb

```diff
@@ -274,6 +274,16 @@ def call(data)
       end
     end
 
+    disambiguate "XML", "Modula-2", "Linux Kernel Module", "AMPL" do |data|
+      if data.include?('<!ENTITY ')
+        Language["XML"]
+      elsif /MODULE\s\w+\s*;/i.match(data) || /^\s*END \w+;$/i.match(data)
+        Language["Modula-2"]
+      else
+        [Language["Linux Kernel Module"], Language["AMPL"]]
```

Comment:
This is the only heuristic that behaves this way. Is there nothing we can do to disambiguate between these two?